### PR TITLE
Update GitParse to handle quoted binary filenames

### LINKED ISSUE #2384: GitParse error: Expected binary line to be in 'Binary files a/fileA and b/fileB differ' format.
Please review the [Community Note](https://github.com/trufflesecurity/trufflehog/blob/main/.github/community_note.md) before submitting

### TruffleHog Version

3.67.1

### Trace Output

```
2024-02-05T13:27:57-05:00       info-2  trufflehog      Expected binary line to be in 'Binary files a/fileA and b/fileB differ' format. {"got": "QmluYXJ5IGZpbGVzIC9kZXYvbnVsbCBhbmQgImIvYXNzZXRzL3JldGFpbGVycy9PTi1pa29ueS1QbGF0Zm9ybWEtZWNvbSBcMzQyXDIwMFwyMjQga29waWEucG5nIiBkaWZmZXIK"}
2024-02-05T13:27:57-05:00       info-2  trufflehog      Expected binary line to be in 'Binary files a/fileA and b/fileB differ' format. {"got": "QmluYXJ5IGZpbGVzIC9kZXYvbnVsbCBhbmQgImIvYXNzZXRzL3JldGFpbGVycy9PTi1pa29ueS1Qcm9zdGEtc2thbGEgXDM0MlwyMDBcMjI0IGtvcGlhLnBuZyIgZGlmZmVyCg=="}
```

### Expected Behavior

The diff is parsed.

### Actual Behavior

The diff is not parsed correctly, because file B is wrapped in quotes:
```
Binary files /dev/null and "b/assets/retailers/ON-ikony-Platforma-ecom \342\200\224 kopia.png" differ
```

### Steps to Reproduce

N/A

## Environment
N/A

## Additional Context

That weird encoding, `\342\200\224`, is apparently an em dash.
https://apple.stackexchange.com/questions/413832/hyphen-is-being-replaced-by-342-200-224-in-bash

### References

N/A

### PR DESCRIPTION
**Pull request by rgmz**
### Description:
This fixes #2384. ~I figured that a regexp would be more robust than checking multiple instances of `bytes.Split`, let me know if you disagree.~

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

**Comment by zricethezav** [comment]
> This fixes https://github.com/trufflesecurity/trufflehog/issues/2384. I figured that a regexp would be more robust than checking multiple instances of bytes.Split, let me know if you disagree.

I would prefer using `bytes.Split` for two reasons: performance (albeit the performance impact of introducing regex here is likely minimal) and simplicity. It _seems_ like it would be easy enough to handle quoted and unquoted cases.

**Comment by rgmz** [comment]
@zricethezav Check it out now.

**Review by zricethezav — COMMENTED** [review]
Thanks for the PR @rgmz, I think it looks good. Appreciate the added test cases as well 👍🏻

File: pkg/gitparse/gitparse.go [hunk 1]
```diff
@@ -708,15 +715,27 @@ func isBinaryLine(latestState ParseState, line []byte) bool {
...
-		return ""
-	}
-	bRaw := sbytes[1]
-	return string(bRaw[:len(bRaw)-8]) // drop the "b/" and " differ\n"
+func pathFromBinaryLine(line []byte) (string, bool) {
+	if bytes.Index(line, []byte("and /dev/null")) != -1 {
+		return "", true
+	}
+
+	_, after, ok := bytes.Cut(line, []byte(" and b/"))
```
[the remark this answers is not in the document]
Comment (rgmz) [reply pkg/gitparse/gitparse.go:723]: Seems like it's actually referring to `bytes.Index`. Feel free to make that change, or I can do it later.

**Review by ahrav — APPROVED** [review]

**Pull request merged**